app/src/main/java/org/jellyfin/androidtv/ui/playback/CustomPlaybackOverlayFragment.java

```diff
@@ -1487,4 +1494,29 @@ private void renderSubtitles(@Nullable final String text) {
             binding.subtitlesText.setVisibility(View.VISIBLE);
         });
     }
+
+    public boolean getServerMeetsMinimumVersion(@NonNull String minServerVersion) {
+        if (sessionRepository.getValue().getCurrentSession().getValue() == null) {
+            Timber.d("failed to get current session");
+            return false;
+        }
+
+        UUID serverId = sessionRepository.getValue().getCurrentSession().getValue().getServerId();
+        AuthenticationStoreServer server = authenticationStore.getValue().getServer(serverId);
+        if (server == null || server.getVersion() == null) {
+            Timber.d("server is null or version is null");
+            return false;
+        }
+
+        ServerVersion version = ServerVersion.Companion.fromString(server.getVersion());
+        ServerVersion minVersion = ServerVersion.Companion.fromString(minServerVersion);
+        if (version == null || minVersion == null) {
+            Timber.d("failed to create ServerVersions from strings: %s %s", version, minVersion);
+            return false;
+        }
+
+        int comparison = version.compareTo(minVersion);
+        Timber.d("current server version [%s] is %s [%s]", version, comparison >= 0 ? ">=" : "<", minVersion);
+        return comparison >= 0;
```

Comment:
In AuthenticationRepository, replace the current getServers function with this:

```kotlin
	private fun AuthenticationStoreServer.asServer(id: UUID) = Server(
		id = id,
		name = name,
		address = address,
		version = version,
		loginDisclaimer = loginDisclaimer,
		dateLastAccessed = Date(lastUsed),
	)

	override fun getServers() = authenticationStore.getServers()
		.map { (id, entry) -> entry.asServer(id) }
		.sortedWith(serverComparator)

	override fun getServer(id: UUID) = authenticationStore.getServer(id)?.asServer(id)
```

(also add `fun getServer(id: UUID): Server?` to the interface, after getServers)

Then, in the server data class, change serverVersion to public or add a function to compare with a ServerVersion instance.

Then get the server via the AuthenticationRepository, or maybe even add a getServer() to the ServerRepository (although that feels a bit redundant..). And use that for the comparison here

Reply:
```patch
diff --git a/app/src/main/java/org/jellyfin/androidtv/auth/AuthenticationRepository.kt b/app/src/main/java/org/jellyfin/androidtv/auth/AuthenticationRepository.kt
index 61cfef7b3..cdfb9556b 100644
--- a/app/src/main/java/org/jellyfin/androidtv/auth/AuthenticationRepository.kt
+++ b/app/src/main/java/org/jellyfin/androidtv/auth/AuthenticationRepository.kt
@@ -31,6 +31,7 @@ import java.util.UUID
 
 interface AuthenticationRepository {
 	fun getServers(): List<Server>
+	fun getServer(id: UUID): Server?
 	fun getUsers(server: UUID): List<PrivateUser>?
 	fun getUser(server: UUID, user: UUID): PrivateUser?
 	fun saveServer(id: UUID, name: String, address: String, version: String?, loginDisclaimer: String?)
@@ -54,16 +55,20 @@ class AuthenticationRepositoryImpl(
 	private val serverComparator = compareByDescending<Server> { it.dateLastAccessed }.thenBy { it.name }
 	private val userComparator = compareByDescending<PrivateUser> { it.lastUsed }.thenBy { it.name }
 
-	override fun getServers() = authenticationStore.getServers().map { (id, info) ->
-		Server(
-			id = id,
-			name = info.name,
-			address = info.address,
-			version = info.version,
-			loginDisclaimer = info.loginDisclaimer,
-			dateLastAccessed = Date(info.lastUsed),
-		)
-	}.sortedWith(serverComparator)
+	private fun AuthenticationStoreServer.asServer(id: UUID) = Server(
+		id = id,
+		name = name,
+		address = address,
+		version = version,
+		loginDisclaimer = loginDisclaimer,
+		dateLastAccessed = Date(lastUsed),
+	)
+
+	override fun getServers() = authenticationStore.getServers()
+		.map { (id, entry) -> entry.asServer(id) }
+		.sortedWith(serverComparator)
+
+	override fun getServer(id: UUID) = authenticationStore.getServer(id)?.asServer(id)
 
 	private fun mapUser(
 		serverId: UUID,
```